### PR TITLE
[SPE-5] Feat: create role assignment UI page

### PR DESCRIPTION
**Pull request by khishh**
### Type of Change

- [x] New Feature
- [ ] Bug Fix
- [ ] Refactoring
- [ ] Optimization
- [ ] Documentation Update
- [ ] DevOps

### Description

Add the initial point for Role Assignment page along with adding react mui dependency.

### Screenshots (optional)

If applicable, add screenshots to help explain your changes. This can be particularly helpful for frontend and design-related changes. Please provide screenshots for different devices and screen sizes (Mobile, Tablet, and Desktop) if possible.

#### Desktop

Desktop Screenshot

#### Tablet

Tablet Screenshot

#### Mobile

Mobile Screenshot

### Deployment Considerations (optional)

If applicable, please provide any deployment considerations or requirements that are necessary for this change, such as database migrations, environment variables, or other required services.

### Checklist (delete irrelevant items)

- [x] I have followed coding standards and best practices.
- [x] My changes generate no new warnings or errors.
- [x] I have manually tested my changes.

### CODE REVIEW

**Review by m-sekander — APPROVED** [review]
LGTM!

File: client/src/views/RoleAssignmentPage.tsx
Comment (YoussefLaunchUp) [inline client/src/views/RoleAssignmentPage.tsx:1]: Please change the folder structure to be consistent with the rest of the project.
https://github.com/LaunchUpCorp/Speakers-corner/pull/3#pullrequestreview-1409997214

**Review by YoussefLaunchUp — CHANGES_REQUESTED** [review]
Good work, small change needed

File: client/src/views/RoleAssignmentPage.tsx
Comment (khishh) [reply client/src/views/RoleAssignmentPage.tsx:1]: Done!

File: client/src/views/RoleAssignmentPage/index.tsx
Comment (YoussefLaunchUp) [inline client/src/views/RoleAssignmentPage/index.tsx:1]: This should be in the page folder not views

**Review by YoussefLaunchUp — CHANGES_REQUESTED** [review]
small change requested

**Review by YoussefLaunchUp — APPROVED** [review]
Great work

**Comment by m-sekander** [comment]
Hey Kodai, looks like you forgot to add the header to the page. For reference, check out my pull request [SPE-14] if you want to see how I used it.

**Review by m-sekander — CHANGES_REQUESTED** [review]
See comment above

**Comment by m-sekander** [comment]
Oh and just want to note if this is a modal, I suppose it wouldn't need the header. Lmk what you think, and I can revert the change request. @khishh

**Review by m-sekander — APPROVED** [review]
LGTM!

**Pull request merged**